pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -242,6 +242,25 @@ func (m *manager) reconcileState() (success []reconciledContainer, failure []rec
 				continue
 			}
 
+			// Check whether container is present in state, there may be 3 reasons why it's not present:
+			// - container is in Be/Bu pod
+			// - kubelet just restarted - state has not been restored properly and is empty
```

Comment:
We changed the way the file-backed state works to panic in this case instead. Can we drop this reason?

Reply:
IMO this is the only valid reason we had to call AddContainer within the reconcile loop in the first place. If it's no longer possible for this to occur we should leave out this entire block.

Reply:
In tryRestoreState() if for some reason we cant parse state file, we create new one (empty) - no panic there (maybe there should be one).
Then starting with new, clean state - in my mind the reconcile loop should be able to pick up and add the running contrariness - i.e. file has been deleted and kubelt restarted without any reconfiguration. (cpuset might get reassigned in that case).

In any case i think that it boils down to a question: Is there any possibility to have running container not present in the state that policy would want to add (in case of `static` policy G pod container) ?

Reply:
> In tryRestoreState() if for some reason we cant parse state file, we create new one (empty) - no panic there (maybe there should be one).
+1 to add panic in this case. 

> In any case i think that it boils down to a question: Is there any possibility to have running container not present in the state that policy would want to add (in case of static policy G pod container) ?
If we make the above change, is there still a chance that a Gu pod container might leak restarts?

Reply:
Panic PR from @ConnorDoyle  #56191